pennylane/ops/op_math/pow.py

```diff
@@ -223,8 +223,26 @@ def label(self, decimals=None, base_label=None, cache=None):
 
     @staticmethod
     def _matrix(scalar, mat):
-        if isinstance(scalar, int) and qml.math.get_deep_interface(mat) != "tensorflow":
-            return qmlmath.linalg.matrix_power(mat, scalar)
+        if isinstance(scalar, int):
+            if qml.math.get_deep_interface(mat) != "tensorflow":
+                return qmlmath.linalg.matrix_power(mat, scalar)
+
+            # TensorFlow doesn't have a matrix_power func, and scipy.linalg.fractional_matrix_power
+            # is not differentiable. So we use a custom implementation of matrix power for integer
+            # exponents below.
+            if scalar == 0:
+                # Used instead of qml.math.eye for tracing derivatives
+                return mat @ qmlmath.linalg.inv(mat)
```

Comment:
I used to just return an identity matrix here before, but in that case the gradient with tf would become `None` rather than 0. Happy to discuss if people don't like this.